File: java_solutions/164.maximum-gap.java
/*
 * @lc app=leetcode id=164 lang=java
 *
 * [164] Maximum Gap
 */

class Solution {
    public int maximumGap(int[] nums) {
        return Optional.of(Arrays.stream(nums).sorted().toArray()).map(sorted -> IntStream.range(0, sorted.length - 1).map(i -> sorted[i + 1] - sorted[i]).max().orElse(0)).get();
    }
}

File: kotlin_solutions/164.maximum-gap.kt
/*
 * @lc app=leetcode id=164 lang=kotlin
 *
 * [164] Maximum Gap
 */

class Solution {fun maximumGap(nums: IntArray): Int = if (nums.size < 2) 0 else nums.sorted().zipWithNext { a, b -> b - a }.maxOrNull() ?: 0}
